### PR TITLE
SNC-520: Add multi oidc tokens with custom claims to docs

### PR DESCRIPTION
**Pull request by gosuku**
# Description
Add a new page for multi oidc tokens custom claims to docs under "Security and Permissions -> Authentication"
Also, add link from the default oidc token overview to the multi oidc token with custom claims doc.

# Reasons
https://circleci.atlassian.net/browse/SNC-520

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

**Review by jbialy — APPROVED** [review]

File: jekyll/_cci2/oidc-tokens-with-custom-claims.adoc [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+contentTags:
+  platform:
+  - Cloud
+---
+= Customize claims in the OIDC token
```
Comment (rosieyohannan) [inline jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:6]:
```suggestion
= On-demand OIDC tokens with custom claims
```

Comment (rosieyohannan) [reply jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:6]: Is this how we want to refer to the new functionality? If this is correct could you also update the filename to match?

File: jekyll/_cci2/oidc-tokens-with-custom-claims.adoc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+:page-layout: classic-docs
+:page-liquid:
+:page-description: Learn about the capabilities to customize claims in OIDC tokens.
+:icons: font
+:experimental:
+
+[#introduction]
+== Introduction
+
```
Comment (rosieyohannan) [inline jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:15]:
```suggestion
```

File: jekyll/_cci2/oidc-tokens-with-custom-claims.adoc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+OpenID Connect (OIDC) tokens allow you to connect your pipelines to compatible cloud services without the need to manage long-lived secrets in CircleCI. CircleCI OIDC tokens can be customized to meet the unique needs of different cloud services.
```
Comment (rosieyohannan) [inline jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:16]:
```suggestion
Use OpenID Connect (OIDC) tokens to connect your pipelines to compatible cloud services without the need to manage long-lived secrets in CircleCI. Create CircleCI OIDC tokens on demand using the CircleCI CLI, and customize your claims to meet the unique needs of different cloud services.
```

File: jekyll/_cci2/oidc-tokens-with-custom-claims.adoc [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+----
+circleci run oidc get --claims '{"aud": "audience_name"}'
+----
+
+The above command will create an OIDC token with the `aud` claim set to "audience_name", and return the token to `stdout`.
+You may pass this value to a variable and use the variable to call the cloud service for authentication.
+
+NOTE: **The OIDC token** created by the above command can be printed with no masking, care should be taken to not log/print the token in the job steps.
+
+Below is an example configuration that deploys a static html webpage to AWS S3 using On-demand OIDC token with customized `aud` claim in a job.
```
Comment (rosieyohannan) [inline jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:35]:
```suggestion

[#example-configuration]
== Example configuration

Below is an example configuration that deploys a static html webpage to AWS S3 using On-demand OIDC token with customized `aud` claim in a job.
```

File: jekyll/_cci2/oidc-tokens-with-custom-claims.adoc [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+:experimental:
+
+[#introduction]
+== Introduction
+
+OpenID Connect (OIDC) tokens allow you to connect your pipelines to compatible cloud services without the need to manage long-lived secrets in CircleCI. CircleCI OIDC tokens can be customized to meet the unique needs of different cloud services.
+See xref:openid-connect-tokens#[Default OpenID Connect tokens in jobs] for more information on default OIDC tokens provided by CircleCI in jobs.
+
+[#on-demand-openid-connect-id-tokens]
+== On demand OpenID Connect ID tokens
```
Comment (rosieyohannan) [inline jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:20]:
```suggestion
[#create-oidc-token-on-demand-in-a-job]
== Create an OIDC token on demand in a job
```

File: jekyll/_cci2/oidc-tokens-with-custom-claims.adoc [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+CircleCI allows creation of OIDC tokens on demand in a job. Mulitple OIDC tokens can be created in a job with options to customize claims.
+The circleci cli may be used in a step to do this.
```
Comment (rosieyohannan) [inline jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:23]:
```suggestion
CircleCI allows creation of OIDC tokens on demand in a job. Multiple OIDC tokens can be created in a job with the option to customize claims. The CircleCI CLI may be used in a step to do this, as follows:
```

File: jekyll/_cci2/oidc-tokens-with-custom-claims.adoc [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+[source,shell]
+----
+circleci run oidc get --claims '{"aud": "audience_name"}'
+----
+
+The above command will create an OIDC token with the `aud` claim set to "audience_name", and return the token to `stdout`.
+You may pass this value to a variable and use the variable to call the cloud service for authentication.
+
+NOTE: **The OIDC token** created by the above command can be printed with no masking, care should be taken to not log/print the token in the job steps.
```
Comment (rosieyohannan) [inline jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:33]:
```suggestion
CAUTION: **The OIDC token** created by the above command can be printed with no masking, care should be taken to not log/print the token in the job steps.
```

File: jekyll/_cci2/oidc-tokens-with-custom-claims.adoc (hunk 4 above)
Comment (rosieyohannan) [inline jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:35]:
```suggestion
Below is an example configuration that deploys a static HTML webpage to AWS S3 using on-demand OIDC token generation, with a customized `aud` claim in a job.
```

File: jekyll/_cci2/oidc-tokens-with-custom-claims.adoc [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+The circleci cli may be used in a step to do this.
+
+[source,shell]
+----
+circleci run oidc get --claims '{"aud": "audience_name"}'
+----
+
+The above command will create an OIDC token with the `aud` claim set to "audience_name", and return the token to `stdout`.
+You may pass this value to a variable and use the variable to call the cloud service for authentication.
+
```
Comment (rosieyohannan) [inline jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:32]: Is it possible to add some kind of list of the various ways claims can be customized in this way? Or is it just `aud`?

Comment (gosuku) [reply jekyll/_cci2/oidc-tokens-with-custom-claims.adoc:32]: I think users can only do `aud` at this time. But we may allow more in the future.

File: jekyll/_data/sidenav.yml [hunk 9]
```diff
@@ -514,6 +514,8 @@ en:
         children:
           - name: Use OpenID Connect tokens in jobs
             link: openid-connect-tokens
+          - name: On demand OIDC tokens with custom claims
+            link: on-demand-oidc-tokens-with-custom-claims
```
Comment (rosieyohannan) [inline jekyll/_data/sidenav.yml:518]:
```suggestion
          - name: OIDC tokens with custom claims
            link: oidc-tokens-with-custom-claims
```

Comment (gosuku) [reply jekyll/_data/sidenav.yml:518]: Thank you for catching this.

File: jekyll/_cci2/oidc-tokens-with-custom-claims.adoc (hunk 1 above)
Comment (gosuku) [reply jekyll/_cci2/oidc-tokens-with-custom-claims.adoc]: We decided to change this to OIDC tokens with custom claims.

**Pull request merged**